src/LedgerId.js

```diff
@@ -0,0 +1,125 @@
+
+const DEFAULT_ERROR = "Default case reached for: ";
+
+export default class LedgerId {
+    /**
+     * @hideconstructor
+     * @internal
+     * @param {Uint8Array} ledgerId
+     */
+    constructor(ledgerId) {
+        /**
+         * @readonly
+         * @type {Uint8Array}
+         */
+        this._ledgerId = ledgerId;
+
+        Object.freeze(this);
+    }
+
+    /**
+     * @param {string} ledgerId
+     * @returns {LedgerId}
+     */
+    static fromString(ledgerId) {
+        switch (ledgerId) {
+            case LedgerId.NETNAMES[0]:
+            case "0":
+                return LedgerId.MAINNET;
+            case LedgerId.NETNAMES[1]:
+            case "1":
+                return LedgerId.TESTNET;
+            case LedgerId.NETNAMES[2]:
+            case "2":
+                return LedgerId.PREVIEWNET;
+            case LedgerId.NETNAMES[3]:
+            case "3":
+                return LedgerId.OTHER;
+            default:
+                throw new Error(
+                    DEFAULT_ERROR.concat("fromString: ledgerId = ").concat(
+                        ledgerId
+                    )
+                );
+        }
+    }
+
+    /**
+     * If the ledger ID is a known value such as `[0]`, `[1]`, `[2]` this method
+     * will instead return "mainnet", "testnet", or "previewnet", otherwise it will
+     * hex encode the bytes.
+     *
+     * @returns {string}
+     */
+    toString() {
+        switch (this._ledgerId[0]) {
+            case 0:
+                return LedgerId.NETNAMES[0];
+            case 1:
+                return LedgerId.NETNAMES[1];
+            case 2:
+                return LedgerId.NETNAMES[2];
+            case 3:
+                return LedgerId.NETNAMES[3];
+            default:
+                return hex.encode(this._ledgerId);
+        }
+    }
+
+    /**
+     * Using the UTF-8 byte representation of "mainnet", "testnet",
+     * or "previewnet" is NOT supported.
+     *
+     * @param {Uint8Array} bytes
+     * @returns {LedgerId}
+     */
+    static fromBytes(bytes) {
+        return new LedgerId(bytes);
+    }
+
+    /**
+     * @returns {Uint8Array}
+     */
+    toBytes() {
+        return this._ledgerId;
+    }
+
+    /**
+     * @returns {boolean}
+     */
+    isMainnet() {
+        return this.toString() == LedgerId.NETNAMES[0];
+    }
+
+    /**
+     * @returns {boolean}
+     */
+    isTestnet() {
+        return this.toString() == LedgerId.NETNAMES[1];
+    }
+
+    /**
+     * @returns {boolean}
+     */
+    isPreviewnet() {
+        return this.toString() == LedgerId.NETNAMES[2];
+    }
+
+    /**
+     * @returns {boolean}
+     */
+    isOther() {
+        return this.toString() == LedgerId.NETNAMES[3];
+    }
+}
+
+LedgerId.MAINNET = new LedgerId(new Uint8Array([0]));
+
+LedgerId.TESTNET = new LedgerId(new Uint8Array([1]));
+
+LedgerId.PREVIEWNET = new LedgerId(new Uint8Array([2]));
+
+LedgerId.OTHER = new LedgerId(new Uint8Array([3]));
```

Comment:
This and `NETNAMES` should not be part of the public API. I'd commend using variables to store these.